python/xorbits/_mars/services/subtask/worker/processor.py

```diff
@@ -341,13 +388,40 @@ async def _store_data(self, chunk_graph: ChunkGraph):
                 storage_level,
             )
             if puts:
+                try:
+                    runner_storage: RunnerStorageActor = await mo.actor_ref(
+                        uid=RunnerStorageActor.gen_uid(self._band[1], self._slot_id),
+                        address=self._supervisor_address, # 这个supervisor_address是不是actor对应的address？
+                    )
+                except mo.ActorNotExist:
+                    # logger.debug(
+                    #     f"Can not find runner storage actor with band name `{self._band}` and slot id `{self._slot_id}",
+                    # )
+                    # self.result.status = SubtaskStatus.errored
+                    # raise
+                    runner_storage: RunnerStorageActor = await mo.create_actor(
+                        RunnerStorageActor,
+                        band=self._band,
+                        slot_id=self._slot_id,
+                        uid=RunnerStorageActor.gen_uid(self._band[1], self._slot_id),
+                        address=self._supervisor_address,
+                        # allocate_strategy=IdleLabel(self._band[1], "storage_runner"),
+                    )
+                # puts 里每个元素都是 DelayedArgument，可用参数 args 取到内部元组 (key,value)
+                for put in puts:
+                    put_key = put.args[0]
+                    put_data = put.args[1]
+                    await runner_storage.put_data(put_key[0] if isinstance(put_key, tuple) else put_key, put_data)
+                
                 put_infos = asyncio.create_task(self._storage_api.put.batch(*puts))
                 try:
                     store_infos = await put_infos
                     for store_key, store_info in zip(stored_keys, store_infos):
                         data_key_to_store_size[store_key] = store_info.store_size
                         data_key_to_memory_size[store_key] = store_info.memory_size
                         data_key_to_object_id[store_key] = store_info.object_id
+                        data_key_to_band[store_key] = self._band
+                        data_key_to_slot_id[store_key] = self._slot_id
```

Comment:
The slot id is not stored to meta?

Reply:
In line 562 of [python/xorbits/_mars/services/subtask/worker/processor.py](https://github.com/xorbitsai/xorbits/pull/728/files/30fac4c618e8e89e8f8fd7f3d7dea36eaf8409a1#diff-338aab235edfe92dd26ebdc6b8efccc59e29d2a029eb133a60245014fb02e079), in `_store_meta()`, we directly added current band and slot_id (`self._band` and `self._slot_id`, specifically). Thus we did not modify the original `_stor_mapper_data` of `SubtaskProcessor` to store mappings from data_key to band or slot_id. Is this still necessary?

Reply:
Meanwhile in file `python/xorbits/_mars/services/storage/handler.py#L639`, StorageHandler calls modified `meta_api.add_chunk_bands`, which needs [slot_id] as additional input, which is not provided by StorageHandlerActor itself. This is still unsolved.